components/datadog/apps/fakeintake/fargateFakeintakeService.go

```diff
@@ -0,0 +1,202 @@
+package fakeintake
+
+import (
+	"fmt"
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/resources/aws"
+	ecsClient "github.com/DataDog/test-infra-definitions/resources/aws/ecs"
+	"github.com/cenkalti/backoff/v4"
+	classicECS "github.com/pulumi/pulumi-aws/sdk/v5/go/aws/ecs"
+	"github.com/pulumi/pulumi-awsx/sdk/go/awsx/awsx"
+	"github.com/pulumi/pulumi-awsx/sdk/go/awsx/ecs"
+	"github.com/pulumi/pulumi-awsx/sdk/go/awsx/lb"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const (
+	oneSecond     = 1000
+	sleepInterval = 1 * oneSecond
+	maxRetries    = 120
+	containerName = "fakeintake"
+	port          = 80
+)
+
+type Instance struct {
+	pulumi.ResourceState
+
+	Host pulumi.StringOutput
+}
+
+func NewECSFargateInstance(e aws.Environment) (*Instance, error) {
+	namer := e.Namer.WithPrefix("fakeintake")
+	opts := []pulumi.ResourceOption{e.WithProviders(config.ProviderAWS, config.ProviderAWSX)}
+
+	instance := &Instance{}
+	if err := e.Ctx.RegisterComponentResource("dd:fakeintake", namer.ResourceName("grp"), instance, opts...); err != nil {
+		return nil, err
+	}
+
+	opts = append(opts, pulumi.Parent(instance))
+
+	var balancerArray classicECS.ServiceLoadBalancerArray
+	var alb *lb.ApplicationLoadBalancer
+	var err error
+	if e.DefaultFargateLoadBalancer() {
+		alb, err = lb.NewApplicationLoadBalancer(e.Ctx, namer.ResourceName("lb"), &lb.ApplicationLoadBalancerArgs{
+			Name:           e.CommonNamer.DisplayName(32, pulumi.String("fakeintake")),
+			SubnetIds:      e.RandomSubnets(),
+			Internal:       pulumi.BoolPtr(!e.ECSServicePublicIP()),
+			SecurityGroups: pulumi.ToStringArray(e.DefaultSecurityGroups()),
+			DefaultTargetGroup: &lb.TargetGroupArgs{
+				Name:       e.CommonNamer.DisplayName(32, pulumi.String("fakeintake")),
+				Port:       pulumi.IntPtr(port),
+				Protocol:   pulumi.StringPtr("HTTP"),
+				TargetType: pulumi.StringPtr("ip"),
+				VpcId:      pulumi.StringPtr(e.DefaultVPCID()),
+			},
+			Listener: &lb.ListenerArgs{
+				Port:     pulumi.IntPtr(port),
+				Protocol: pulumi.StringPtr("HTTP"),
+			},
+		}, opts...)
+		if err != nil {
+			return nil, err
+		}
+
+		instance.Host = alb.LoadBalancer.DnsName()
+		balancerArray = classicECS.ServiceLoadBalancerArray{
+			&classicECS.ServiceLoadBalancerArgs{
+				ContainerName:  pulumi.String(containerName),
+				ContainerPort:  pulumi.Int(port),
+				TargetGroupArn: alb.DefaultTargetGroup.Arn(),
+			},
+		}
+	} else {
+		var FargateErr error
+		instance.Host, FargateErr = FargateServiceFakeintake(e)
+		if FargateErr != nil {
+			return nil, FargateErr
+		}
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
No need to create a variable here
```suggestion
		instance.Host, err = FargateServiceFakeintake(e)
		if err != nil {
			return nil, err
		}
```